Compiler/src/typelattice.jl

```diff
@@ -318,20 +318,23 @@ end
         fields = vartyp.fields
         thenfields = thentype === Bottom ? nothing : copy(fields)
         elsefields = elsetype === Bottom ? nothing : copy(fields)
+        undef = copy(vartyp.undef)
         for i in 1:length(fields)
             if i == fldidx
                 thenfields === nothing || (thenfields[i] = thentype)
                 elsefields === nothing || (elsefields[i] = elsetype)
+                undef[i] = false
             end
         end
```

Comment:
```suggestion
        if 1 ≤ fldidx ≤ length(fields)
            thenfields === nothing || (thenfields[fldidx] = thentype)
            elsefields === nothing || (elsefields[fldidx] = elsetype)
            undef[fldidx] = false
        end
```
My previous implement seems quite silly..